tools/osquery-agent-options/main.go

```diff
@@ -171,3 +222,77 @@ func camelCaseOptionName(s string) string {
 	}
 	return strings.Join(parts, "")
 }
+
+// sanitizeArchivePath sanitizes the archive file pathing from "G305: Zip Slip vulnerability"
+func sanitizeArchivePath(d, t string) (string, error) {
+	v := filepath.Join(d, t)
+	if strings.HasPrefix(v, filepath.Clean(d)) {
+		return v, nil
+	}
+
+	return "", fmt.Errorf("%s: %s", "content filepath is tainted", t)
+}
+
+// extractTagGz extracts the contents of the provided tar.gz file.
+func extractTarGz(path string) error {
```

Comment:
It doesn't make sense to rewrite this standard yet somewhat complex functionality here. Can you use the same extractTarGz method as the `update` package?

Reply:
It differs in that the one in `update` uses `pkg/secure` methods.

_Sometimes_ (like in this scenario for tooling) code duplication is a-ok IMO :) (https://x.com/ID_AA_Carmack/status/753745532619665408)